otelcli/plumbing.go

```diff
@@ -19,8 +19,93 @@ import (
 	"google.golang.org/grpc/credentials"
 )
 
+// initTracer sets up the OpenTelemetry plumbing so it's ready to use.
+// Returns a context and a func() that encapuslates clean shutdown.
+func initTracer() (context.Context, func()) {
+	ctx := context.Background()
+
+	otel.SetErrorHandler(diagnostics)
+
+	// when no endpoint is set, do not set up plumbing. everything will still
+	// run but in non-recording mode, and otel-cli is effectively disabled
+	// and will not time out trying to connect out
+	if config.Endpoint == "" {
+		return ctx, func() {}
+	}
+
+	var exporter sdktrace.SpanExporter // allows overwrite in --test mode
+	var err error
+
+	// The OTel spec does not support grpc:// or any way to deterministically demand
+	// gRPC via the endpoint URI, preferring OTEL_EXPORTER_PROTOCOL to do so. This is
+	// awkward for otel-cli so we break with the spec. otel-cli will only resolve
+	// http(s):// to HTTP protocols, defaults bare host:port to gRPC, and supports
+	// grpc:// to definitely use gRPC to connect out.
```

Comment:
This func didn't change much except for this part. I was annoyed it was below the helpers it calls (they were added in a contributed PR a while ago) so I moved it back to the top.

Reply:
looks like no functional change at all, just added the above comment and moved the function